### PR TITLE
deps: move from @npmcli/ci-detect to ci-info

### PR DESCRIPTION
**Pull request by wraithgar**
See https://github.com/watson/ci-info/pull/95 for more context on
achieving parity between these two modules.

This changes npm to use `ci-info` instead of `@npmcli/ci-detect`.
Everything that npm currently flags as a CI environment should still be
doing so, so there is no breaking change there.

There is going to be a subtle difference in the `ci-name` config, which
nothing in npm currently looks at anyways, as well as the ci name that
shows up in the default `user-agent` string.  Some providers will be
slightly different (i.e. circle-ci vs circleci and cirrus vs cirrus-ci)

### CODE REVIEW

File: lib/utils/config/definitions.js [hunk 1]
```diff
@@ -433,7 +432,7 @@ define('cert', {
 })
 
 define('ci-name', {
-  default: ciName || null,
+  default: ciInfo.name ? ciInfo.name.toLowerCase().split(' ').join('-') : null,
```
Comment (nlf) [inline lib/utils/config/definitions.js:435]: nit: i think this could be simplified to `ciInfo.name?.toLowerCase().split(' ').join('-')`

**Review by nlf — APPROVED** [review]
one very minor nit you can take or leave, but this looks good otherwise

File: lib/utils/config/definitions.js (hunk 1 above)
Comment (wraithgar) [reply lib/utils/config/definitions.js:435]: we'd still need a `|| null` cause `?.` returns undefined

Comment (nlf) [reply lib/utils/config/definitions.js:435]: ah fair, so we need this to be specifically `null` and not just anything false-y. disregard my comment then

File: lib/utils/config/definitions.js [hunk 2]
```diff
@@ -442,7 +441,7 @@ define('ci-name', {
   description: `
     The name of a continuous integration system.  If not set explicitly, npm
     will detect the current CI environment using the
-    [\`@npmcli/ci-detect\`](http://npm.im/@npmcli/ci-detect) module.
+    [\`ci-info\`](http://npm.im/@npmcli/ci-info) module.
```
Comment (sibiraj-s) [inline lib/utils/config/definitions.js:444]: Invalid URL for ci-info. `http://npm.im/ci-info`.

Also here https://github.com/npm/cli/pull/5858/files#diff-6f21b9547f08121788002c1b5494afcd450c3d6a9a6f4e1f452d71d2764515b4R761

Comment (wraithgar) [reply lib/utils/config/definitions.js:444]: fiddlesticks.  search and replace failed me, sorry about that.

this file is the source of truth, changing it and running snapshots will fix the rest.

**Pull request merged**